src/converse-mam-views.js

```diff
@@ -40,17 +40,30 @@ converse.plugins.add('converse-mam-views', {
                  * the last archived message in our local cache.
                  */
                 if (this.disable_mam) { return; }
+
                 const { _converse } = this.__super__,
                       most_recent_msg = u.getMostRecentMessage(this.model);
 
                 if (_.isNil(most_recent_msg)) {
                     this.fetchArchivedMessages();
                 } else {
                     const stanza_id = most_recent_msg.get(`stanza_id ${this.model.get('jid')}`);
+                    const that = this;
+
+                    const success = function(messages) {
+                        /* If we received exactly the maximum number of messages, there's a good
+                        chance more messages are available. Let's run this method again to check. */
+                        if (_.size(messages) == _converse.archived_messages_page_size) {
+                            /* must use setTimeout here; otherwise the MAM messages we added just now are
```

Comment:
The XMPP server will inform you when the results are paged, so it should be necessary to do this here.
See the (relevant part of the XEP](https://xmpp.org/extensions/xep-0313.html#query-limit).

Specifically you should check for a `<set xmlns='http://jabber.org/protocol/rsm>` element and then use that to determine whether you need to query for more pages.